extensions/ql-vscode/src/helpers.ts

```diff
@@ -266,6 +269,36 @@ export function isFolderAlreadyInWorkspace(folderName: string) {
   );
 }
 
+/** Check if the current workspace is the CodeTour and open the workspace folder.
+ * Without this, we can't run the code tour correctly.
+ **/
+export async function prepareCodeTour(): Promise<void> {
+  if (workspace.workspaceFolders?.length) {
+    const currentFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+    // We need this path to check that the file exists on windows
+    const tutorialWorkspacePath = join(
+      currentFolder,
+      "tutorial.code-workspace",
+    );
+    const toursFolderPath = join(currentFolder, ".tours");
+
+    if (
+      existsSync(tutorialWorkspacePath) &&
+      existsSync(toursFolderPath) &&
```

Comment:
Could these be async? Otherwise you have a mix of sync and async operations in this async handler.

Reply:
It looks the async versions of `exists`, `access` and `stat` all expect something to happen in a callback, rather than returning a boolean so they're a bit unwieldy:

```
exists('/path/to/a/file', (e) => {
  console.log(e ? 'it exists' : 'no file found!');
});
```

Other example: https://github.com/github/vscode-codeql/blob/3c229d244e53eb4cd7df93abff1c48d6b2352a1f/extensions/ql-vscode/src/test-adapter.ts#L330

I'm happy to adapt it though, just not sure what we've done so far in the extension for these cases. There seem to be multiple solutions.

Reply:
Thanks! This seems to work fine 👍